### ISSUE TITLE
panic when pulling images

### ISSUE THREAD

**Issue by aluzzardi** [docker/swarm#326]
```
panic: runtime error: slice bounds out of range

goroutine 43 [running]:
runtime.panic(0x7ad8c0, 0xa0e48f)
	/usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
main.(*logHandler).Handle(0xa13008, 0xc2083decc0, 0x0, 0x0)
	/root/go/src/github.com/docker/swarm/manage.go:26 +0x446
github.com/docker/swarm/cluster.(*Cluster).Handle(0xc20801b310, 0xc2083decc0, 0x0, 0x0)
	/root/go/src/github.com/docker/swarm/cluster/cluster.go:74 +0xce
github.com/docker/swarm/cluster.(*Node).handler(0xc2082dcb60, 0xc20841bd80, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/docker/swarm/cluster/node.go:463 +0x143
github.com/docker/swarm/cluster.*Node.(github.com/docker/swarm/cluster.handler)·fm(0xc20841bd80, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/docker/swarm/cluster/node.go:102 +0x58
github.com/samalba/dockerclient.(*DockerClient).getEvents(0xc2081102e0, 0xc20831a600, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/samalba/dockerclient/dockerclient.go:264 +0x5b9
created by github.com/samalba/dockerclient.(*DockerClient).StartMonitorEvents
	/root/go/src/github.com/samalba/dockerclient/dockerclient.go:243 +0x83
```

The offender seems to be:
```
	log.WithFields(log.Fields{"node": e.Node.Name, "id": e.Id[:12], "from": e.From, "status": e.Status}).Debug("Event received")
```

`e.Id` in this case was `redis`, resulting in out of bound panic.

**Comment by aluzzardi** [comment]
This was the offender:

```
DEBU[0139] Event received                                from= id=redis node=ubuntu-1 status=pull
```

**Comment by vieux** [comment]
fixed by #328
